src/config/outputs/shell/container.rs

```diff
@@ -0,0 +1,87 @@
+use std::rc::Rc;
+
+pub use calloop::channel::Sender;
+use egui::{CtxRef, Ui};
+use rhai::{plugin::*, INT};
+
+pub use crate::config::eventloop::ConfigEvent;
+
+use super::panel::{Panel as PanelContainer, PanelPosition};
+use super::r#box::Box as BoxContainer;
+use super::widget::Widget;
+
+#[derive(Debug, Clone)]
+pub enum Layout {
+    Vertical,
+    Horizontal,
+}
+
+impl Layout {
+    pub fn render(
+        &self,
+        ui: &mut Ui,
+        widgets: &Vec<Rc<dyn Widget>>,
```

Comment:
If this Vec is immutable anyway, and is used only for iteration, we can just do:
```suggestion
        widgets: &[Rc<dyn Widget>],
```

Reply:
sure, that makes more sense, changed :)